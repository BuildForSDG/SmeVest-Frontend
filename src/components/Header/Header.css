.Header {
  height: 95px;
  box-shadow: 0 1px 3px rgba(0, 0, 0, 0.1);
  font-family: 'Karla', sans-serif;
  background: #fff;
}

.Header a {
  text-decoration: none;
}

.Header a > img {
  width: 80px;
  height: auto;
  max-width: 100%;
}

.Header .NavigationList {
  display: flex;
  justify-content: space-between;
  background: #fff;
}

.Header .NavigationList a {
  font-size: 0.95rem;
  padding: 0.3rem 1.5rem;
  transition: all 300ms linear;
}

.Header .SignIn {
  color: #928d8d;
}

.Header .SignIn:hover {
  color: #212226;
  text-decoration: underline;
  text-decoration-color: #212226;
}

.Header .SignUp {
  background: #4ad6e9;
  border-radius: 1rem;
  color: #fff;
  box-shadow: 0 2px 1px #1fc2d8, 0 -1px 1px #1fc2d8;
  border: 1px solid #4ad6e9;
}

.Header .Dashboard {
  background: #4ad6e9;
  border-radius: 1rem;
  color: #fff;
  box-shadow: 0 2px 1px #1fc2d8, 0 -1px 1px #1fc2d8;
  border: 1px solid #4ad6e9;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.Header .SignUp:hover,
.Header .Dashboard:hover {
  background: transparent;
  color: #212226;
  box-shadow: 0 0px 0px rgba(0, 0, 0, 0.7), 0px 0px #fff;
  text-decoration: none;
}

.Header .SignOut {
  border-radius: 1rem;
  color: #fff;
  display: flex;
  align-items: center;
  color: #d4475a;
}

@media screen and (max-width: 768px) {
  .Header {
    padding-bottom: 6rem !important;
  }

  .Header .NavigationList {
    padding: 1.6rem 0;
  }

  .Header .SignIn,
  .Header .Dashboard {
    margin-bottom: 0.5rem;
  }

  .Header .Dashboard {
    justify-content: flex-start;
  }
}
