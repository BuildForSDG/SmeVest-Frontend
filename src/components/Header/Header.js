import React from 'react';
import { Link } from 'react-router-dom';
import { Nav, Navbar, Container } from 'react-bootstrap';
import { Dashboard, ExitToApp } from '@material-ui/icons';
import './Header.css';
import logo from '../../assets/images/logo.png';

const loggedIn = false;
const Header = () => (
  <Navbar className="Header" expand="xl">
    <Container>
      <Navbar.Brand>
        <Link to="/">
          <img src={logo} alt="Logo" className="logo" />
        </Link>
      </Navbar.Brand>
      <Navbar.Toggle aria-controls="basic-navbar-nav" />
      <Navbar.Collapse id="basic-navbar-nav">
        <Nav className="ml-auto NavigationList">
          {loggedIn ? (
            <>
              <Link to="/dashboard" className="Dashboard">
                <Dashboard /> Dashboard
              </Link>
              <Link to="/signout" className="SignOut">
                <ExitToApp /> Sign Out
              </Link>
            </>
          ) : (
            <>
              <Link to="/signin" className="SignInLink">
                Sign In
              </Link>
              <Link to="/signup" className="SignUpLink">
                Sign Up
              </Link>
            </>
          )}
        </Nav>
      </Navbar.Collapse>
    </Container>
  </Navbar>
);

export default Header;
