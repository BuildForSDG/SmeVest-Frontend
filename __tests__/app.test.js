import React from 'react';
import { shallow } from 'enzyme';
import App from '../src/App.js';
import Header from '../src/components/Header/Header';

describe('<App /> Component', () => {
  test('render App component', () => {
    expect(shallow(<App />).length).toEqual(1);
  });

  test('should contain Header Component', () => {
    expect(shallow(<App />).find(Header).length).toEqual(1);
  });

  test('should take a snapshot of App component', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  });
});
