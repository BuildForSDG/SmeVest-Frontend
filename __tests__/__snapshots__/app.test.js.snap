// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Component should take a snapshot of App component 1`] = `ShallowWrapper {}`;
